### PR TITLE
Add a feature flag around /host/etc detectio

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

When running in a container with /etc mounted to /host/etc, we use that
folder for OS detection. Configuration 'ignore_host_etc' allow to
disable this behavior if needed.

### Motivation

There is no real reason why a user would mount `/host/etc` and would not want for the Agent to use it. But since this is a change in behavior is safer to have a feature flag around it.

### Describe how to test/QA your changes

Run the Agent in a container of a OS for the container different that the host. Mount the host `/etc` to `/host/etc` in the container.

The host OS should be detected by the Agent by default. Then set `ignore_host_etc` to `true` and check that the container OS is detected.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by paulcacheux** [comment]
Hello ! The `HOST_ETC` env var is also used in this function https://github.com/DataDog/datadog-agent/blob/6d4ba1332757d08a23e315f5cfd27c799ba51d9f/pkg/config/system_probe.go#L169

Should this change as well ?

**Comment by hush-hush** [comment]
> Hello ! The `HOST_ETC` env var is also used in this function
> 
> https://github.com/DataDog/datadog-agent/blob/6d4ba1332757d08a23e315f5cfd27c799ba51d9f/pkg/config/system_probe.go#L169
> 
> Should this change as well ?

Not really. My change is about forcing `HOST_ETC` if we detect `/host/etc` is mounted. We want to force it so the rest of the code and side library (like gopsutil) can rely on it. If the user manually set `HOST_ETC` then we will use it.

This feature is basically an helper for user. Before they had to mount the folder and set the env var, now they can just mount the folder.

File: pkg/config/config.go [hunk 1]
```diff
@@ -1143,6 +1132,24 @@ func findUnknownEnvVars(config Config) []string {
 	return unknownVars
 }
 
+func useHostEtc(config Config) {
+	if IsContainerized() && pathExists("/host/etc") {
+		if !config.GetBool("ignore_host_etc") {
```
Comment (remeh) [inline pkg/config/config.go:1137]: We could consider doing this config first, instead of after a `IsContainerized()` or `pathExists` call, WDTY?

Comment (hush-hush) [reply pkg/config/config.go:1137]: I wanted to print a debug message when `ignore_host_etc` is enabled and has an impact (ie: containerized+/host/etc)

**Review by remeh — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

**Pull request merged**